src/biogeochem/CNPhenologyMod.F90

```diff
@@ -2048,13 +2099,119 @@ subroutine CropPhenology(num_pcropp, filter_pcropp                     , &
                hui(p) = max(hui(p),huigrain(p))
             endif
 
+            do_harvest = .false.
+            force_harvest = .false.
+            fake_harvest = .false.
+            sown_today = .false.
+            if (use_cropcal_rx_sdates .and. s > 0) then
+                sown_today = crop_inst%sdates_thisyr(p,s) == real(jday, r8)
+            end if
+
+            ! TEMPORARY? GGCMI seasons often much longer than CLM mxmat.
+            mxmat = pftcon%mxmat(ivt(p))
+            if (use_mxmat .and. generate_crop_gdds) then
+               call endrun(msg="If setting generate_crop_gdds to .true., you must set use_mxmat to .false.")
+            else if (.not. use_mxmat) then
+                mxmat = 999
+            end if
+
+            if (jday == 1 .and. croplive(p) .and. idop(p) == 1 .and. sowing_count(p) == 0) then
+                ! Crop was incorrectly planted in last time step of Dec. 31.
+                do_harvest = .true.
+                force_harvest = .true.
+                fake_harvest = .true.
+                harvest_reason = 3._r8
+            else if (do_plant .and. .not. did_plant) then
+                ! Today was supposed to be the planting day, but the previous crop still hasn't been harvested.
+                do_harvest = .true.
+                force_harvest = .true.
+                harvest_reason = 4._r8
```

Comment:
It works because only one planting is allowed per year. See the first condition here:
```fortran
 if (sowing_count(p) == mxsowings) then
	do_plant_normal = .false.
	do_plant_lastchance = .false.
 else if (ivt(p) == nwwheat .or. ivt(p) == nirrig_wwheat) then
	! winter temperate cereal : use gdd0 as a limit to plant winter cereal
	!
	! Are all the normal requirements for planting met?
	do_plant_normal = allow_unprescribed_planting           .and. &
			  a5tmin(p)   /= spval                  .and. &
			  a5tmin(p)   <= minplanttemp(ivt(p))   .and. &
			  jday        >= minplantjday(ivt(p),h) .and. &
			  jday        <= maxplantjday(ivt(p),h) .and. &
			  ...
	...
...
```

In the future, if multiple cropping is enabled by the use of multiple sowing windows, we'd need to add another dimension to `[min,max]plantjday` so that they're patch x hemisphere x **window**, and index them like `minplantjday(ivt(p), h, sowing_count(p)+1)`.